src/main/kotlin/ecommerce/CollectionProductStore.kt

```diff
@@ -0,0 +1,42 @@
+package ecommerce
+
+import ecommerce.exception.NotFoundException
+import ecommerce.model.Product
+import org.springframework.stereotype.Repository
+
+@Repository
+class CollectionProductStore(val products: MutableList<Product>) : ProductStore {
+    override fun countProducts(): Int {
+        return products.size
+    }
+
+    override fun findAll(): List<Product> {
```

Comment:
Since this is only used in tests, you can move it under the test package.
Also, since it doesn't use Spring's container for bean injection, you can remove the annotations as well.

```suggestion
class CollectionProductStore(
    private val products: MutableList<Product>,
) : ProductStore {
    override fun countProducts(): Int {
        return products.size
    }

    override fun findAll(): List<Product> {
```